software/py/vanilla_stats_parser.py

```diff
@@ -0,0 +1,472 @@
+    self.manycore_dim = manycore_dim_y * manycore_dim_x
+    self.per_tile_stat = per_tile_stat
+
+
+    self.max_tile_groups = 1024
+    self.num_tile_groups = 0
+
+    self.timing_start_list = [0] * self.max_tile_groups 
+    self.timing_end_list =   [0] * self.max_tile_groups
+#    self.timing_stat = []
+
+    self.total_execution_time = 0
+    self.total_instr_cnt = 0
+    self.total_stall_cnt = 0
+    self.total_miss_cnt = 0
+
+    self.tile_stat_dict = [[dict() for x in range(self.manycore_dim_x)] for y in range(self.manycore_dim_y)]
+    self.manycore_stat_dict = dict()
+
+
+
+  # private method
+  # parse the stats file to generate total execution time
+  # for each tile group and for total
+  def generate_stats_timing(self, trace):
+    y = trace["y"]
+    x = trace["x"]
+    relative_y = y - bsg_TILE_GROUP_ORG_Y
+    relative_x = x - bsg_TILE_GROUP_ORG_X
+    tg_num = trace["tag"]
+    # Only count those coming from the origin tile (0,0)
+    if (relative_x == 0 and relative_y == 0):
+      if(tg_num < bsg_PRINT_STAT_END_VAL):
+        self.timing_start_list[tg_num] = trace["time"]
+        self.num_tile_groups += 1
+      else:
+        self.timing_end_list[tg_num - bsg_PRINT_STAT_END_VAL] = trace["time"]
+
+
+  # private method
+  # Sum up all other stats for all tiles based on the last bsg_print_stat instr
+  def generate_stats_all(self):
+      # Generate timing stats 
+      for trace in self.traces:
+        self.generate_stats_timing(trace)
+
+      self.timing_stat = [0] * self.num_tile_groups
+      for tg_id in range (0, self.num_tile_groups):
+        self.timing_stat[tg_id] = self.timing_end_list[tg_id] - self.timing_start_list[tg_id];
+
+
+      #other stats are only read once per tile from the end of file
+      #i.e. if mesh dimensions are 4x4, only last 16 lines are needed 
+      trace_idx = len(self.traces)
+      for y in range(self.manycore_dim_y):
+        for x in range(self.manycore_dim_x):
+          trace_idx -= 1
+          trace = self.traces[trace_idx]
+          for stat in stats_list:
+            self.tile_stat_dict[y][x][stat] = trace[stat]
+          for instr in instr_list:
+            self.tile_stat_dict[y][x][instr] = trace[instr]
+          for stall in stalls_list:
+            self.tile_stat_dict[y][x][stall] = trace[stall]
+          for miss in miss_list:
+            self.tile_stat_dict[y][x][miss] = trace[miss]
+
+
+  # private method
+  # print execution timing for the entire manycore 
+  def print_manycore_stats_timing(self, stat_file):
+    # For total execution time, we only sum up the execution time of origin tile in tile groups
+    # The origin tile's relative coordinates is always 0,0 
+    stat_file.write("Timing Stats\n")
+    stat_file.write(stats_timing_header_format.format("tile group", "exec time", "share (%)"))
+    stat_file.write("{}".format(separating_line))
+
+    # Sum up execution time for all tile groups 
+    for i in range (0, self.num_tile_groups):
```

Comment:
I think you can just call `sum(self.timing_stat)`

Reply:
Good point!